lib/samples/filter_by_definition_expression_or_display_filter/filter_by_definition_expression_or_display_filter_sample.dart

```diff
@@ -27,23 +27,27 @@ class FilterByDefinitionExpressionOrDisplayFilterSample extends StatefulWidget {
 
 class _FilterByDefinitionExpressionOrDisplayFilterSampleState
     extends State<FilterByDefinitionExpressionOrDisplayFilterSample> {
+  // create a map view controller
   final _mapViewController = ArcGISMapView.createController();
-  final _featureLayer = FeatureLayer.withFeatureTable(
+  final FeatureLayer _featureLayer = FeatureLayer.withFeatureTable(
       ServiceFeatureTable.withUri(Uri.parse(
           'https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/SF_311_Incidents/FeatureServer/0')));
 
   @override
   void initState() {
     super.initState();
-
+    // create a map with a topographic basemap style
```

Comment:
This is a good opportunity to discuss this setup pattern that we talked about last week. We should probably establish a pattern for what we consider best practice in terms of setting up the map so that it doesn't raise questions with those using our samples as a guide. It will also come up when we implement tutorials etc. on the developer website.

In most of the samples we've done the last few weeks we've used `_onMapViewReady` to do configuration. Sometimes that's because we need the map view to actually be ready before we can do "stuff".
It feels to me like it's good to use `initState` if we can - but then should we be using `initState` for everything that can be done up front, and then use `onMapViewReady` for anything that requires the map view to be instantiated? An alternative that avoids separation of setup code and keeps the sample a little more linear could be to say that if we need `onMapViewReady` then set everything up there, as we have been doing, and if not use `initState`, like this sample.

@PaulAllanSturm I'd be interested in your thoughts here.

Reply:
> Which implies that we abandon initState(). It's not bad or wrong, but only onMapViewReady() allows us to be consistent across samples for all use cases.

Sounds good. And I completely agree that samples often need compromise to be readable vs hyper-optimized.

So @shokhinaToj , for this sample, could you move the code in `initState` into `onMapViewReady`? And if you want to go through anything in the above discussion, just give me a shout.

Reply:
I suppose it should say, "..., if possible". Definitely you can't avoid `late` in all cases.

One situation I'm thinking of is something like this:
```dart
final _graphicsOverlay = GraphicsOverlay()
  ..renderer = SimpleRenderer(symbol: defaultMarkerSymbol)
  ..graphics.addAll([firstGraphic, secondGraphic, thirdGraphic]);
```

What I'm getting at is, we don't want all that stuff jumbled up where the variable is declared, but also we don't want to declare it `late GraphicsOverlay _graphicsOverlay`.

So my idea is, it's declared like this:
```dart
final _graphicsOverlay = GraphicsOverlay();
```
... and in onMapViewReady, it gets the rest of the setup work done:
```dart
void onMapViewReady() async {
  _graphicsOverlay.renderer = SimpleRenderer(symbol: defaultMarkerSymbol)
  ..graphics.addAll([firstGraphic, secondGraphic, thirdGraphic]);
  // ...
}
```

Maybe I should phrase it: "If possible, avoid declaring a member variable `late`. Prefer instead to use the constructor when the variable is declared. But note that any other setup work should be done in `onMapViewReady()`."

Reply:
Ohhh now it makes sense!